pkg/apis/rsyslog/validation/rsyslog_relp.go

```diff
@@ -112,6 +139,19 @@ func validateLoggingRules(loggingRules []rsyslog.LoggingRule, fldPath *field.Pat
 	return allErrs
 }
 
+func validateProgramNames(programNames []string, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	for index, name := range programNames {
+		if invalidCharactersForProgramNameRegex.MatchString(name) {
```

Comment:
Sorry for forgetting to add this to the previous review. Could we also forbid program  names that are empty strings?